client/src/main/java/io/pravega/client/netty/impl/FlowHandler.java

```diff
@@ -266,12 +271,14 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
     @Override
     public void close() {
         if (closed.compareAndSet(false, true)) {
-            Channel ch = channel.get();
+            Channel ch = channel.getAndSet(null);
             if (ch != null) {
                 log.debug("Closing channel {} ", ch);
                 final int openFlowCount = flowIdReplyProcessorMap.size();
                 if (openFlowCount != 0) {
                     log.warn("{} flows are not closed", openFlowCount);
```

Comment:
I'm wondering if this log message makes sense because I believe we are closing the flows with the next line, is it right?

Reply:
I have improved the logging now.